File: lab2-functionalprimer/python3/func_as_arg.py
def listFunc(a, b):
    return [i for i in range(a, b + 1)]

def applicatorFunc(inpFunc, a, b, s):
    if s =='s':
        return sum(inpFunc(a, b))
    else:
        return sum(inpFunc(a, b)) / (b - a + 1)

print(applicatorFunc(listFunc, 1, 5, 'a'))

File: lab2-functionalprimer/haskell/func_as_arg.hs
--create inpFunc
inpFunc a b = [a..b]

--Define applicatorFunc
applicatorFunc a b s = if s =='s' then sum (inpFunc a b) else sum (inpFunc a b) / (b - a + 1)

main = do
    let result = applicatorFunc 1 5 'a'
    putStrLn("sum = " ++ show(result))